backend/src/main/java/com/bakdata/conquery/apiv1/query/CQElement.java

```diff
@@ -87,6 +87,6 @@ public void visit(Consumer<Visitable> visitor) {
 	}
 
 	public RequiredEntities collectRequiredEntities(QueryExecutionContext context) {
-		return new RequiredEntities(context.getBucketManager().getEntities().keySet());
+		return new RequiredEntities(context.getBucketManager().getEntity2Bucket().keySet());
```

Comment:
kannst du das Pattern bitte kapseln in getBucketManager -> getEntities? Vlt sogar noch in readonly Set-wrappen, damits nicht ausversehen knallt.

Reply:
Habs gekapselt